stem_cell/differentiator.py

```diff
@@ -1,18 +1,30 @@
 class Differentiator:
     PAYLOAD_ENTRY_ANSIBLE_PLAYBOOK = '/main.yml'
 
-    def __init__(self, user_data_gateway, version_control_system, ansible):
+    def __init__(self, user_data_gateway, version_control_system, ansible, file_gateway):
         self.ansible = ansible
+        self.file_gateway = file_gateway
         self.user_data_gateway = user_data_gateway
         self.version_control_system = version_control_system
 
     def differentiate(self):
-        payload_location = self.clone_differentiation_payload()
+        payload_location = self.clone_payload()
         self.ansible.run_playbook(payload_location + self.PAYLOAD_ENTRY_ANSIBLE_PLAYBOOK)
 
-    def clone_differentiation_payload(self):
+    def clone_payload(self):
+        private_key = self.user_data_gateway.get_version_control_private_key()
+        if private_key:
+            return self.clone_with_private_key(private_key)
+
+        return self.clone_with_url()
+
+    def clone_with_url(self):
         return self.version_control_system.clone(self.get_version_control_url())
 
+    def clone_with_private_key(self, key):
+        self.file_gateway.write_file('/tmp/key.pem', key)
+        return self.version_control_system.clone_with_private_key(self.get_version_control_url(), '/tmp/key.pem')
```

Comment:
I'm a little rusty on Python, though I do recall that similar to Ruby, there's a community aversion to letting Java's get/set method prefix convention creep in